docs/proposals/coap-core.md

```diff
@@ -0,0 +1,74 @@
+# [CoAPCore](https://tools.ietf.org/html/rfc6690#:~:text=well-known%2Fcore) Protocol Implementation
+
+## Goal
+
+_From the RFC 6690_:
+
+The Constrained RESTful Environments (CoRE) realizes the Representational State Transfer (REST) architecture [REST] in a suitable form for the most constrained nodes (e.g., 8-bit microcontrollers with limited memory) and networks (e.g., IPv6 over Low-Power Wireless Personal Area Networks (6LoWPANs) [RFC4919]). CoRE is aimed at Machine-to-Machine (M2M) applications such as smart energy and building automation.
+
+The main function of such a discovery mechanism is to provide Universal Resource Identifiers (URIs, called links) for the resources hosted by the server, complemented by attributes about those resources and possible further link relations. In CoRE, this collection of links is carried as a resource of its own (as opposed to HTTP headers delivered with a specific resource).
+
+## Discovery Process
+
+### The standard
+
+A well-known relative URI `/.well-known/core` is defined as a default entry point for requesting the list of links about resources hosted by a server and thus performing CoRE Resource Discovery. The CoRE protocol is applicable for use with Constrained Application Protocol (CoAP) [COAP](https://tools.ietf.org/html/rfc7252), HTTP, or any other suitable web transfer protocol.
+
+Resource Discovery can be performed either unicast or multicast.
+
+When a server’s IP address is already known, either a priori or resolved via the Domain Name System (DNS), unicast discovery is performed in order to locate the entry point to the resource of interest. In this specification, this is performed using a GET to `/.well-known/core` on the server, which returns a payload in the CoRE Link Format.
+
+Multicast Resource Discovery is useful when a client needs to locate a resource within a limited scope, and that scope supports IP multicast. A GET request to the appropriate multicast address is made for `/.well-known/core`. In order to limit the number and size of responses, a query string is recommended with the known attributes. Typically, a resource would be discovered based on its Resource Type and/or Interface Description, along with possible application-specific attributes.
+
+### With Akri
+
+The Akri Configuration defines a list of IP addresses to use for resource discovery, implementing thus unicast discovery. Adding support to multicast discovery is surely desiderable.
```

Comment:
A CoAP broadcast request is just an IP multicast, the increase of difficulty lies in handling responses from multiple devices but it shouldn't be hard. I already planned to consider this feature as the highest priority. This would allow discovery of every device within the same network.

Unfortunately I'll have less time to work on akri in the next days because I'm moving to other tasks in my thesis, but I'll give a try asap :)

Reply:
Awesome! And no worries. We will still be here.

Reply:
I've added support to multicast today 🎉